### DISCUSSION TITLE
adding a mail server programatically without an api

### DISCUSSION THREAD

**Discussion by ergineroglu** [postalserver/postal#2188] [Help with using Postal]
Hi,

I have checked other discussions about an api to add domains or create mail servers:

- https://github.com/postalserver/postal/pull/992
- https://github.com/postalserver/postal/issues/82 

I understand it will take time to implement and maintain a new api, so what I am asking for is can we add a new mail server and its configuration via adding the configuration into directly into database and just trigger postal to load new records from the database?

I have also checked for a discussion about database schema to figure out the details but couldn't find one.

**Comment by willpower232** [comment]
Unfortunately for this case, creating a mailserver requires creating a new database within the database server so it would need to be done via Postals code to ensure that the database migrations were run and all tables present and correct.

It would be best to wait for @adamcooke 's new api version

**Reply by ergineroglu** [reply]
thank you for the clarification Will

**Reply by janiskelemen** [reply]
@willpower232 @adamcooke, thanks for open sourcing such a great project! Do you know when the API version will be available? Is there anything we can help to push this? Happy to support the development with funds if that helps! :)
